### DISCUSSION TITLE
How do I get my bot to look for messages that include a phrase

### DISCUSSION THREAD

**Discussion by mangoess** [PrismarineJS/mineflayer#2705] [Q&A] [answered]
This is just an example of my code

    switch (message) {
      case '!order':
        bot.chat("you have no life")
        break
Can anyone help me on how I can make it so that if anybody says !order (anything else) it will still run the command, thanks!

**Comment by ichenglin** [comment, marked as answer]
The **`chat event`** from **mineflayer** pass in a **player's chat message**,
so all we have to do here is **match it with regex** pattern **`^!order (.*)$`**.

The **example** below would do **`console.log()`** when **player's message** matches **`!order <any string>`**,

```js
bot.on("chat", (username, message, translate, jsonMsg, matches) => {
    const message_matcher = message.match(/^!order (.*)$/);
    if (!message_matcher || !message_matcher[1]) {
        // message doesn't match
        return;
    }
    console.log(`${username} executed "!order" command with parameters "${message_matcher[1]}"`)
});
```

**Reply by mangoess** [reply]
Thank you very much!
